spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -937,18 +921,12 @@ def setup_editor(self,
         self.set_auto_unindent_enabled(auto_unindent)
         self.set_indent_chars(indent_chars)
 
-        # Show/hide the debug panel depending on the language and parameter
-        self.set_debug_panel(show_debug_panel, language)
```

Comment:
As far as I can tell from the master, this is already the case. When looking for `show_debug_panel` in the codebase, I can't find any place it is called (only from `setup_editor` but then `setup_editor` is never called with `show_debug_panel` argument)

Reply:
Maybe an issue should be opened but I don't think this is a regression of this PR

Reply:
We're using `set_debug_panel` to hide the debug panel (now `BreakpointsManager`) when the language is not Python (that's implemented in method you removed below).

I think you should reimplement that code in the `_add_codeeditor` method of the `Debugger` plugin, so that that panel is not unconditionally added to all CodeEditor's but only to the ones that display Python files.

Reply:
Done